recipes/cminpack/all/conanfile.py

```diff
@@ -0,0 +1,109 @@
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools import files
+from conan import ConanFile
+import os
+import textwrap
+
+required_conan_version = ">=1.45.0"
+
+
+class CMinpackConan(ConanFile):
+    name = "cminpack"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "About A C/C++ rewrite of the MINPACK software (originally in FORTRAN)" \
+                  "for solving nonlinear equations and nonlinear least squares problems"
+    topics = ("nonlinear", "solver")
+    homepage = "http://devernay.free.fr/hacks/cminpack/"
+    license = "BSD-like"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_EXAMPLES"] = "OFF"
+        tc.variables["CMINPACK_LIB_INSTALL_DIR"] = "lib"
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def layout(self):
+        cmake_layout(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+        # cminpack is a c library
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def source(self):
+        files.get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
+
+    def build(self):
+        cmake = CMake(self)
+        # moving this into the toolchain variables is not sufficient
+        # right now. required for setting shared option
+        cmake.configure(variables={
+            "CMAKE_POLICY_DEFAULT_CMP0077":  "NEW"
+        })
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+
+        files.rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        files.copy(self, "CopyrightMINPACK.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        files.rmdir(self, os.path.join(self.package_folder, "share")) # contains cmake config files
+
+    def library_postfix(self):
```

Comment:
```suggestion
    def _library_postfix(self):
```
Any custom method should be protected, to avoid any possible name collision with future conan features

Reply:
Good point!